src/main/java/se/liu/ida/hefquin/engine/queryproc/impl/optimizer/evolutionaryAlgorithm/InitializeNrGeneration.java

```diff
@@ -0,0 +1,47 @@
+package se.liu.ida.hefquin.engine.queryproc.impl.optimizer.evolutionaryAlgorithm;
+
+import se.liu.ida.hefquin.engine.queryplan.LogicalOperator;
+import se.liu.ida.hefquin.engine.queryplan.LogicalPlan;
+import se.liu.ida.hefquin.engine.queryplan.logical.impl.*;
+
+public class InitializeNrGeneration {
+
+    public static int countNumOfOp( final LogicalPlan plan ) {
+        final LogicalOperator lop = plan.getRootOperator();
+
+        if ( lop instanceof LogicalOpMultiwayJoin) {
+            final int count = plan.numberOfSubPlans();
+
+            int nrGeneration = calcFactorial(count);
```

Comment:
Sorry I didn't say it earlier, but I realize it only now: you can simplify the code by combining the cases of LogicalOpMultiwayJoin and LogicalOpMultiwayUnion into a single if-block. The only thing that you then need to be careful of is how you initialize `nrGeneration` -- it needs to be as follows:
```
int nrGeneration = ( lop instanceof LogicalOpMultiwayJoin ) ? calcFactorial(count) : 1;
```